cmd/contour/serve.go

```diff
@@ -294,6 +296,36 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 		XffNumTrustedHops:             ctx.Config.Network.XffNumTrustedHops,
 	}
 
+	if ctx.Config.RateLimitService.ExtensionService != "" {
+		namespacedName := k8s.NamespacedNameFrom(ctx.Config.RateLimitService.ExtensionService)
+		client := clients.DynamicClient().Resource(contour_api_v1alpha1.ExtensionServiceGVR).Namespace(namespacedName.Namespace)
+
+		// ensure the specified ExtensionService exists
+		res, err := client.Get(context.Background(), namespacedName.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("error getting rate limit extension service %s: %v", namespacedName, err)
```

Comment:
Should this be a fatal error or not configure rate limiting until the extension service exists?

Reply:
Agree it's not ideal to require the extension service to exist at startup. I think we could change the flow somewhat to resolve the extension service ref during the DAG build instead. If it's OK with you, I'll file a follow-up issue to look at that?

Reply:
yup sure thing!